Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2923,11 +2923,20 @@ extension BrowserViewController: PreferencesObserver {
             // All `block all cookies` toggle requires a hard reset of Webkit configuration.
             tabManager.reset()
             if !Preferences.Privacy.blockAllCookies.value {
-                tabManager.allTabs.forEach {
-                    if let url: URL = $0.webView?.url {
-                        $0.loadRequest(PrivilegedRequest(url: url) as URLRequest)
+                HTTPCookie.loadFromDisk { (_) in
+                    self.tabManager.reloadSelectedTab()
+                    self.tabManager.allTabs.forEach {
```

Comment:
You could change it to `for tab in self.tabManager.allTabs where self.tabManager.selectedTab != tab {`
to remove the `if` inside of the loop. Up to you if you find your way more readable